dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -110,45 +110,50 @@ public GHRepository getRepo(String repoName) throws IOException {
+            terms.forEach(search::q);
+            PagedSearchIterable<GHContent> files = search.list();
+            int totalCount = files.getTotalCount();
+            log.info("Number of files found for {}: {}", image, totalCount);
+            if (totalCount > gitApiSearchLimit
+                    && orgsToIncludeOrExclude.size() == 1
+                    && orgsToIncludeOrExclude
                     .entrySet()
                     .stream()
                     .findFirst()
                     .get()
-                    .getKey();
-            log.warn("Number of search results for a single org {} is above {}! The GitHub Search API will only return around 1000 results - https://developer.github.com/v3/search/#about-the-search-api",
-                    orgName, gitApiSearchLimit);
-        } else if (totalCount > gitApiSearchLimit) {
-            log.info("The number of files returned is greater than the git API search limit"
-                    + " of {}. The orgs with the maximum number of hits will be recursively removed"
-                    + " to reduce the search space. For every org that is excluded, a separate "
-                    + "search will be performed specific to that org.", gitApiSearchLimit);
-            return getSearchResultsExcludingOrgWithMostHits(image, files, orgsToIncludeOrExclude, gitApiSearchLimit);
+                    .getKey() != null
+                    && orgsToIncludeOrExclude
+                    .entrySet()
+                    .stream()
+                    .findFirst()
+                    .get()
+                    .getValue()
+            ) {
+                String orgName = orgsToIncludeOrExclude
+                        .entrySet()
+                        .stream()
+                        .findFirst()
+                        .get()
+                        .getKey();
+                log.warn("Number of search results for a single org {} is above {}! The GitHub Search API will only return around 1000 results - https://developer.github.com/v3/search/#about-the-search-api",
+                        orgName, gitApiSearchLimit);
+            } else if (totalCount > gitApiSearchLimit) {
+                log.info("The number of files returned is greater than the git API search limit"
+                        + " of {}. The orgs with the maximum number of hits will be recursively removed"
+                        + " to reduce the search space. For every org that is excluded, a separate "
+                        + "search will be performed specific to that org.", gitApiSearchLimit);
+                return getSearchResultsExcludingOrgWithMostHits(image, files, orgsToIncludeOrExclude, gitApiSearchLimit);
+            }
+            List<PagedSearchIterable<GHContent>> filesList = new ArrayList<>();
+            filesList.add(files);
+            return Optional.of(filesList);
+        } catch (GHException | HttpException e) {
```

Comment:
I think it is best to keep this method as is and catch these exceptions in the `All` class. The reason is that now there is side-effect behavior on our Parent sub-command, which should be expected to fail if we find an image that cannot be looked up.